.buildkite/steps/test-demo-notebooks.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+set -xeo pipefail
+
+stellargraph_dir="$PWD"
+
+SPLIT="${BUILDKITE_PARALLEL_JOB_COUNT:-1}"
+INDEX="${BUILDKITE_PARALLEL_JOB:-0}"
+
+echo "--- :books: collecting notebooks"
+# Create array with all notebooks in demo directories.
+cd "${stellargraph_dir}"
+NOTEBOOKS=()
+while IFS= read -r -d $'\0'; do
+  NOTEBOOKS+=("$REPLY")
+done < <(find "${stellargraph_dir}/demos" -name "*.ipynb" -print0)
+
+NUM_NOTEBOOKS_TOTAL=${#NOTEBOOKS[@]}
+
+if [ "$NUM_NOTEBOOKS_TOTAL" -ne "$SPLIT" ]; then
+  echo "Buildkite step parallelism is set to $SPLIT but found $NUM_NOTEBOOKS_TOTAL notebooks. Please update parallelism to match the number of notebooks!"
+  exit 1
+fi
+
+echo "--- installing dependencies"
+pip install -q --no-cache-dir '.[demos]'
+
+echo "--- listing dependency versions"
+pip freeze
+
+f=${NOTEBOOKS[$INDEX]}
+
+echo "+++ :python: running $f"
+case $(basename "$f") in
+  'attacks_clustering_analysis.ipynb' | 'hateful-twitters-interpretability.ipynb' | 'hateful-twitters.ipynb' | 'stellargraph-attri2vec-DBLP.ipynb' | \
+    'node-link-importance-demo-gat.ipynb' | 'node-link-importance-demo-gcn.ipynb' | 'node-link-importance-demo-gcn-sparse.ipynb' | 'rgcn-aifb-node-classification-example.ipynb' | \
+    'stellargraph-metapath2vec.ipynb' | \
+    'ensemble-node-classification-example.ipynb' | 'cora-gcn-links-example.ipynb')
+    # FIXME: these notebooks do not yet work on CI (#818 - datasets can't be downloaded)
```

Comment:
Totally minor, but the convention in the rest of the library at the moment is to put the issue number next to the FIXME (this makes grepping mildly easier, and ensures that the number is close to the `FIXME` text even if the rest of the line is long and wrapped), e.g.

```python
# The notebooks do not yet work on CI:
# FIXME (#816): missing dataset download
# FIXME (#818): datasets can't be downloaded
# FIXME (#819): out-of-memory
# FIXME (#833): too slow
```